lib/improver/tests/utilities/test_cube_metadata.py

```diff
@@ -129,18 +130,123 @@ def test_update_ukv(self):
+        already exist."""
+        changes = {
+            "units": {
+                "action": "set",
+                "units": "K"
+                }
+            }
+        cube = self.cube.copy()
+        cube.units = None
+        expected = self.cube.copy()
+        result = update_units(cube, changes)
+        self.assertEqual(result, expected)
+
+    def test_cube_overwrite(self):
+        """Modify the units on the cube by overwriting any existing units."""
+        changes = {
+            "units": {
+                "action": "set",
+                "units": "Celsius"
+                }
+            }
+        cube = self.cube.copy()
+        expected = self.cube.copy()
+        expected.units = Unit("Celsius")
+        result = update_units(cube, changes)
+        self.assertEqual(result, expected)
+
+    def test_cube_convert(self):
+        """Modify the units on the cube by converting any existing units."""
+        changes = {
+            "units": {
+                "action": "convert",
+                "units": "Celsius"
+                }
+            }
+        cube = self.cube.copy()
+        expected = self.cube.copy()
+        expected.convert_units(Unit("Celsius"))
+        result = update_units(cube, changes)
+        self.assertEqual(result, expected)
+
+    def test_coord_add(self):
+        """Modify the units on the coordinate by adding units where they do not
+        already exist."""
+        changes = {
+            "units": {
+                "action": "set",
+                "units": "seconds since 1970-01-01 00:00:00"
+                }
+            }
+        coord = self.cube.coord("time").copy()
+        coord.units = None
+        expected = self.cube.coord("time").copy()
+        result = update_units(coord, changes)
+        self.assertEqual(result, expected)
+
+    def test_coord_convert(self):
+        """Modify the units on the coordinate by overwriting any
+        existing units."""
+        changes = {
+            "units": {
+                "action": "convert",
+                "units": "hours since 1970-01-01 00:00:00"
+                }
+            }
+        coord = self.cube.coord("time").copy()
+        expected = self.cube.coord("time").copy()
+        expected.convert_units(Unit("hours since 1970-01-01 00:00:00"))
+        result = update_units(coord, changes)
+        self.assertEqual(result, expected)
+
+    def test_unknown_action(self):
+        """Check an error is raised if the specified action is not known."""
+        changes = {
+            "units": {
+                "action": "unknown",
+                "units": "K"
+                }
+            }
+        coord = self.cube.coord("time").copy()
+        msg = "The specified action is not known"
+        with self.assertRaisesRegex(KeyError, msg):
+            update_units(coord, changes)
+
+
 class Test_add_coord(IrisTest):
 
     """Test the add_coord method."""
 
+    def setUp(self):
```

Comment:
If you're doing this, would you mind also factoring out `cube = create_cube_with_threshold()`, as this is duplicated throughout this test class.

Reply:
I've done this.